src/ToolInfo.php

```diff
@@ -26,67 +26,92 @@
     const COMPOSER_INSTALLED_FILE = '/../../composer/installed.json';
     const COMPOSER_PACKAGE_NAME = 'friendsofphp/php-cs-fixer';
 
-    public static function getComposerVersion()
+    /**
+     * @var string
+     */
+    private $script;
+
+    /**
+     * @var null|string
+     */
+    private $composerVersion;
+
+    /**
+     * @var null|bool
+     */
+    private $isInstalledAsPhar;
+
+    /**
+     * @var null|bool
+     */
+    private $isInstalledByComposer;
+
+    /**
+     * @var null|string
+     */
+    private $scriptDir;
+
+    /**
+     * @param string $script
+     */
+    public function __construct($script)
     {
```

Comment:
The perfect final form of this constructor, imho, would it to be `$this->script = $script;` and somewhere else a factory that injects the real script name, and factories for classes that use it that inject `ToolInfo` through its factory.
In the meantime...